tests/dialects/test_arith.py

```diff
@@ -244,3 +244,10 @@ def test_cmpi_incorrect_comparison():
         # 'oeq' is a comparison op for cmpf but not cmpi
         _cmpi_op = Cmpi.get(a, b, "oeq")
     assert e.value.args[0] == "Unknown comparison mnemonic: oeq"
+
+
+def test_cmpi_index_type():
+    a = Constant.from_int_and_width(1, IndexType())
```

Comment:
Could you add such a test to MLIR conversion as well? Just to make sure we catch it, if they decide to change anything around that...